### ISSUE TITLE
ImportError: cannot import name 'dataset_pb2'

### ISSUE THREAD

**Issue by Rosywind** [waymo-research/waymo-open-dataset#48]
When I run "from waymo_open_dataset import dataset_pb2 as open_dataset", there would report an error "ImportError: cannot import name 'dataset_pb2'". 
What is the reason for this? What is the 'dataset_pb2'?
Looking forward to your reply！

**Comment by peisun1115** [comment]
Does 
pip3 install waymo-open-dataset==1.0.1

 work for you?

If not, what is your python version, tensorflow version?

**Comment by yklose** [comment]
Have the same issue. The pip install does not work, it shows the error:

ERROR: Could not find a version that satisfies the requirement waymo-open-dataset==1.0.1 (from versions: none)
ERROR: No matching distribution found for waymo-open-dataset==1.0.1

Python Version:              3.6.5 
Tensorflow Version:        1.9

Thanks in advance.

**Comment by peisun1115** [comment]
we require tensorflow 1.14

**Comment by Gus-Guo** [comment]
Hi, I also met the same error"ERROR: Could not find a version that satisfies the requirement waymo-open-dataset (from versions: none)
ERROR: No matching distribution found for waymo-open-dataset"
my pip version is 19.3.1. Do I need anything else? I think tensorflow is not needed for now. Thank you very much.

**Comment by peisun1115** [comment]
The pre-built pip package requires tensorflow 1.14.0.  @Gus-Guo, do you prefer to not installing tf?

**Comment by peisun1115** [comment]
Another way to do it:
https://github.com/waymo-research/waymo-open-dataset/issues/35

Alternatively, you can also copy the files after compiling locally:

waymo-od is the directory you downloaded the source: 
cp waymo-od/bazel-genfiles/waymo_open_dataset/label_pb2.py waymo-od/waymo_open_dataset/label_pb2.py
cp waymo-od/bazel-genfiles/waymo_open_dataset/dataset_pb2.py waymo-od/waymo_open_dataset/dataset_pb2.py

**Comment by yklose** [comment]
Thank you, #35 solved my problem!

**Comment by Gus-Guo** [comment]
> The pre-built pip package requires tensorflow 1.14.0. @Gus-Guo, do you prefer to not installing tf?

Thank you very much. I change the python verison to from 3.7 to 3.6 and it works.

**Comment by Rosywind** [comment]
> Does
> pip3 install waymo-open-dataset==1.0.1
> 
> work for you?
> 
> If not, what is your python version, tensorflow version?

Yes, the log had show "Successfully installed ... waymo-open-dataset-1.0.1 ...". I found the problem is about the installation directory. Thank you very much.

**Comment by abhishupatel** [comment]
Hello @peisun1115,
I am working on a project which uses your dataset. Currently I have windows 10 and I am using Jupyter Notebook to import your dataset. I am facing hardship in trying to install the waymo_open_dataset using pip3. 
I have 
Python Version = 3.6.8
Tensorflow version = 1.14.0

I am geting the following error.
![image](https://user-images.githubusercontent.com/57020689/67608339-a6cc0e00-f73c-11e9-8700-3452d5226bfe.png)

I tried a workaround by cloning your waymo-open-dataset and then changing the directory to it and all imports started working out but then in frame_utils you are using dataset_pb2 which I do not have in the cloned git repo so it is throwing error.

Can you guys please help me out in either instaling the waymo-open-dataset or importing the dataset_pb2. Also, is it possible to run everything on jupyter notebook on windows 10 or is it a requirement to have linux OS??

**Comment by pwais** [comment]
`pip3 install waymo-open-dataset==1.0.1 --no-deps` if you don't want waymo to clobber your tensorflow install (or lack thereof)

**Comment by WilliamCabrera** [comment]
Hello Guys, my name is William, i'm an student at Sherbrooke University in Quebec and my research project is related to autonomous drivingn so i will be using Waymo-open-dataset. I was trying to install it by following the guide:  https://colab.research.google.com/github/waymo-research/waymo-open-dataset/blob/v1.0.1/tutorial/tutorial.ipynb#scrollTo=hDQ1DPqwdfNW but it does not work at all, it give the same error that abhishupatel  has. Is there any other way to install it? I have the same python and tensorflow version than abhishupatel   has. Could anybody help me with this?

**Comment by peisun1115** [comment]
@WilliamCabrera 

What is the error you are seeing? 

If you just want to get started without compiling, try 

pip3 install upgrade --pip
pip3 install waymo-open-dataset-tf-2-0-0==1.0.1 --user

https://github.com/waymo-research/waymo-open-dataset/blob/master/docs/quick_start.md

**Comment by WilliamCabrera** [comment]
Hi Pei Sun,
Thank you very much for your answer but i'm still getting the same error:

cmd logs:

(yolo_cpu) D:\PersonalProjetcs\Tensorflow>pip3 install
waymo-open-dataset-tf-2-0-0==1.0.1 --user
ERROR: Can not perform a '--user' install. User site-packages are not
visible in this virtualenv.

(yolo_cpu) D:\PersonalProjetcs\Tensorflow>pip3 install
waymo-open-dataset-tf-2-0-0==1.0.1
ERROR: Could not find a version that satisfies the requirement
waymo-open-dataset-tf-2-0-0==1.0.1 (from versions: none)
ERROR: No matching distribution found for waymo-open-dataset-tf-2-0-0==1.0.1

**Comment by peisun1115** [comment]
@WilliamCabrera Ah, you are using windows. 

That is not officially supported for now.  Either use a linux machine or use a docker container or just copy the code you need from this repo for your own purpose.

**Comment by WilliamCabrera** [comment]
Hi Pei Sun,

Do you know where can i find this on the python code: dataset_pb2?
according to the documentation that is the way of opening a frame file.

FILENAME = '/content/waymo-od/tutorial/frames'
dataset = tf.data.TFRecordDataset(FILENAME, compression_type='')
for data in dataset:
    frame = open_dataset.Frame()
    frame.ParseFromString(bytearray(data.numpy()))
    break
Thanks a lot for your help i really appreciate it Pei Sun.

Regards,
William

**Comment by peisun1115** [comment]
@WilliamCabrera 

You can generate that on your own with [protoc](http://google.github.io/proto-lens/installing-protoc.html).

Alternatively, generate that in the docker container we provide here by running bazel build.

Or just use [this](https://github.com/gdlg/simple-waymo-open-dataset-reader/tree/master/simple_waymo_open_dataset_reader) which is contributed by the community.  This is not officially supported and it can be out of date.

**Comment by WilliamCabrera** [comment]
Hello Pei Sun,

I finally run the waymo demo  scripts and i want to thank you for your
help. I would like to ask you something related to the format of the
tfrecord files.

1 tfrecord(800mb) = 5 frames(5 cameras) + lidar points ?

I tried to  run a sequence but i noticed that the next tfrecord file is not
the next step in the sequence, i mean it is an image of something
completely different. I loaded the two tfrecord files i that i marked on
the screenshot i attached in this email and the are completely different .
I would like to know if it is possible with waymo dataset to load a set of
tfrecords where each  tfrecords  represents the next frame  in a video
sequence, so i could try to simulate a real-time detection over a video.

Kind regards,
William

**Comment by peisun1115** [comment]
They are consecutive. Did you check whether they are from the same camera (by checking the camera name)? Did you check their frame.timestamp_micros to see whether they are approximately 0.1 sec apart?

**Comment by WilliamCabrera** [comment]
this is the second image. There are sorted according to the number on the
tfreccord file name. I did not check the frame.timestamp_micros since it
says invalid date when i tried to check the date but load the next(3rd
file) with same camera and it give another image also different.

**Comment by jvlmdr** [comment]
@Gus-Guo I had the same problem: the `waymo-open-dataset` package does not have a wheel for python 3.7. If you instead install the package `waymo-open-dataset-tf-2-1-0` as described in the "quick start" page, there is a wheel for python 3.7.

**Comment by peisun1115** [comment]
@WilliamCabrera I cannot see the images in your replies.

@jvlmdr we have updated our quick start. https://github.com/waymo-research/waymo-open-dataset/blob/master/docs/quick_start.md#use-pre-compiled-pippip3-packages

**Comment by J-Rojas** [comment]
I'm unable to run `pip3 install waymo-open-dataset=1.0.1` or `pip3 install waymo-open-dataset-tf-1-15-0==1.2.0`. 

I've tried on Mac and Linux (ubuntu 18.04)

```
ERROR: Could not find a version that satisfies the requirement waymo-open-dataset (from versions: none)
ERROR: No matching distribution found for waymo-open-dataset
```

**Comment by monghimng** [comment]
For anyone who is just playing with the data on MacOS (thus cannot just use pip3 install), as of May 17, 2020, to make this work all you need to put the two files from [here](https://github.com/Yao-Shao/Waymo_Kitti_Adapter/tree/master/protocol%20buffer) into the directory `waymo_open_dataset`. This way, the imports will be able to find `dataset_pb2` as per the colab tutorial.

As peisun1115 noted above, this could be outdated in the future.

**Comment by hughlee815** [comment]
> > Does
> > pip3 install waymo-open-dataset==1.0.1
> > work for you?
> > If not, what is your python version, tensorflow version?
> 
> Yes, the log had show "Successfully installed ... waymo-open-dataset-1.0.1 ...". I found the problem is about the installation directory. Thank you very much.

hello, what does  _the problem is about the installation directory_mean? i got the same problem as yours.
i have tried:
-python 3.6
-tensorflow 2.1.0
and
-python 3.6 
-tensorflow 1.14.0
but neither nor woks for me. please help.

**Comment by praveenphatate** [comment]
> > > Does
> > > pip3 install waymo-open-dataset==1.0.1
> > > work for you?
> > > If not, what is your python version, tensorflow version?
> > 
> > 
> > Yes, the log had show "Successfully installed ... waymo-open-dataset-1.0.1 ...". I found the problem is about the installation directory. Thank you very much.
> 
> hello, what does _the problem is about the installation directory_mean? i got the same problem as yours.
> i have tried:
> -python 3.6
> -tensorflow 2.1.0
> and
> -python 3.6
> -tensorflow 1.14.0
> but neither nor woks for me. please help.

Did you figure it out I am having the same issue on Windows Conda

**Comment by WilliamCabrera** [comment]
Hello Pei Sun, could you please help me find out how to access the 3D
bounding for each image? I locked on the Frame object but i did not find
anything.

Regards,
William

**Comment by ZhaohanJackWang** [comment]
Hi there, I am unable to install Waymo dataset using !pip3 install waymo-open-dataset-tf-2-1-0==1.2.0. My python version is 3.7.6 and my Tensorflow version is 2.1.0, which I believe are sufficient. 

The error I got is: 
ERROR: Could not find a version that satisfies the requirement waymo-open-dataset-tf-2-1-0==1.2.0
ERROR: No matching distribution found for waymo-open-dataset-tf-2-1-0==1.2.0

Does anyone know what might caused this? 
Thanks in advance.
<img width="1107" alt="Screen Shot 2021-03-13 at 6 16 15 pm" src="https://user-images.githubusercontent.com/66340428/111023053-d7444300-842a-11eb-9ee4-1fdb6931ca52.png">

**Comment by peisun1115** [comment]
Try 1.3.0?

**Comment by ZhaohanJackWang** [comment]
Hi Pei Sun,
I tried 1.3.0, but still not working. 
The error is:
ERROR: Could not find a version that satisfies the requirement waymo-open-dataset-tf-2-1-0==1.3.0
ERROR: No matching distribution found for waymo-open-dataset-tf-2-1-0==1.3.0

**Comment by peisun1115** [comment]
Consider updating your pip?

**Comment by ZhaohanJackWang** [comment]
I think I've already updated pip using: !pip3 install --upgrade pip, in the first step
I believe my command is identical to the tutorial
<img width="1065" alt="Screen Shot 2021-03-13 at 7 03 56 pm" src="https://user-images.githubusercontent.com/66340428/111023754-f2b14d00-842e-11eb-86fb-60ff22763cb4.png">

**Comment by peisun1115** [comment]
What is your OS? 

You can also try the provider docker file

**Comment by ZhaohanJackWang** [comment]
Hi Pei Sun, my system is macOS version10.14.6. I am now trying docker but still encountered some issues building the docker container. 
<img width="1371" alt="Screen Shot 2021-03-14 at 3 29 23 pm" src="https://user-images.githubusercontent.com/66340428/111057265-3f9e2d80-84da-11eb-83e9-d64a6e0468e3.png">

**Comment by SHRIVP** [comment]
Hi I am able to install everything and run through the final steps of the colab tutorial https://github.com/waymo-research/waymo-open-dataset/blob/master/tutorial/tutorial_motion.ipynb tutorial.
However, while learning the simple MLP model of the last step of the tutorial I am getting the below error 

ImportError: cannot import name 'motion_metrics_pb2' from 'waymo_open_dataset.protos' (/usr/local/lib/python3.7/dist-packages/waymo_open_dataset/protos/__init__.py)

I looked at the above location and don't see any module as motion_metrics_pb2 I do see motion_metrics.I was successfully able to import this file but looks like this is not the correct file as I get the below error :  

module 'waymo_open_dataset.protos.metrics_pb2' has no attribute 'MotionMetricsConfig'

**Comment by iyuner** [comment]
> > > Hi I am able to install everything and run through the final steps of the colab tutorial https://github.com/waymo-research/waymo-open-dataset/blob/master/tutorial/tutorial_motion.ipynb tutorial.
> > > However, while learning the simple MLP model of the last step of the tutorial I am getting the below error
> > > ImportError: cannot import name 'motion_metrics_pb2' from 'waymo_open_dataset.protos' (/usr/local/lib/python3.7/dist-packages/waymo_open_dataset/protos/**init**.py)
> > > I looked at the above location and don't see any module as motion_metrics_pb2 I do see motion_metrics.I was successfully able to import this file but looks like this is not the correct file as I get the below error :
> > > module 'waymo_open_dataset.protos.metrics_pb2' has no attribute 'MotionMetricsConfig'
> > 
> > 
> > Hello,
> > I have the same problem. Did you figure out how to solve this?
> 
> use this!
> 
> !pip3 install waymo-open-dataset-tf-2-3-0

This works! Thanks! I follow the [tutorial_motion.ipynb](https://github.com/waymo-research/waymo-open-dataset/blob/master/tutorial/tutorial_motion.ipynb), and the error shows python version is 3.7. According to the [quick_start](https://github.com/waymo-research/waymo-open-dataset/blob/master/docs/quick_start.md), for `waymo-open-dataset-tf-2-1-0` mentioned in the ipynb, it available for python 3.8. So change the version which available for 3.7 makes it work!

**Comment by pramenku** [comment]
[2021-09-23T10:27:30.899Z] ERROR: Could not find a version that satisfies the requirement waymo-open-dataset-tf-2-6-0 (from versions: none)

[2021-09-23T10:27:30.899Z] ERROR: No matching distribution found for waymo-open-dataset-tf-2-6-0

Getting this issue on CentOS7, SLES 15 and Ubuntu18.04.5

**Comment by peisun1115** [comment]
what is your python version? we  have only pre-built py 7, 8, 9. for other versions, you can build on your own

https://pypi.org/project/waymo-open-dataset-tf-2-6-0/#files

**Comment by pramenku** [comment]
> what is your python version? we have only pre-built py 7, 8, 9. for other versions, you can build on your own
> 
> https://pypi.org/project/waymo-open-dataset-tf-2-6-0/#files

python3.6 
You meant to use Python 3.7/3.8/3.9 to install waymo-open-dataset-tf-2-6-0?

**Comment by peisun1115** [comment]
Yes, we no longer pre-build python 3.6 since 1.4.1 . we can add it back though if py3.6 is  still popular.

but it is really easy to build this on your own if you have docker installed. 

https://github.com/waymo-research/waymo-open-dataset/tree/master/pip_pkg_scripts

in summary: either upgrade your python or build on your own

**Comment by pramenku** [comment]
> Yes, we no longer pre-build python 3.6 since 1.4.1 . we can add it back though if py3.6 is still popular.
> 
> but it is really easy to build this on your own if you have docker installed.
> 
> https://github.com/waymo-research/waymo-open-dataset/tree/master/pip_pkg_scripts
> 
> in summary: either upgrade your python or build on your own

Thanks.  I am not sure how many has switched to Py3.7/3.8/3.9.  I guess Py3.6 is still very popular python version of 3.x series.
Can we make a version which works on all 3.x version, Kind of backward compatible.?

For now, I can use Python 3.8 as this is 2nd most used 3.x version after 3.6

**Comment by peisun1115** [comment]
that is a good point. i can probably add py3.6 back in a few days.

**Comment by pramenku** [comment]
> that is a good point. i can probably add py3.6 back in a few days.

Thanks for the help. Hoping, by 27th Sept, we can have it ready. :)

**Comment by Pradeep1902** [comment]
> What is your OS?
> 
> You can also try the provider docker file

I am using windows 10 OS and I have the similar issue. Does it not work on windows?
